### PR TITLE
Refactor Time methods 

### PR DESCRIPTION
**Pull request by straight-shoota**
This PR refactors `Time.year_month_day_year` and `Time.absolute_days` to simplify the code and make it easier to understand.

The changes are mostly just formal, but there are a few semantic changes such as storing the value of an expression in a local variable for re-use.

### CODE REVIEW

File: src/time.cr [hunk 1]
```diff
@@ -1216,11 +1220,12 @@ struct Time
...
-    m = 1
-
-    days = DAYS_MONTH
-    totaldays = offset_seconds / SECONDS_PER_DAY
+  # Returns the calendrical representation of this instance's date.
+  #
+  # The return value is a tuple consisting of year (`1..9999`), month (`1..12`),
+  # day (`1..31`) and ordinal day of the year (`1..366`).
+  protected def year_month_day_day_year : {Int32, Int32, Int32, Int32}
+    totaldays = (offset_seconds / SECONDS_PER_DAY).to_i
```
Comment (asterite) [inline src/time.cr]: Maybe name it `total_days`?

Comment (straight-shoota) [reply src/time.cr]: I had considered that, but wasn't sure about it. Will change it :+1:

**Comment by straight-shoota** [comment]
`year_month_day_day_year` is about 25% faster than before.

I actually didn't expect much change in `absolute_days` but for some reason it's about 10% slower. But the measurements vary a lot, it's hard to nail it down, but that seems to be the trend.
When I change the operations back exactly how it was before, it somehow measures about 10% improvement.
This is all really weird and inconclusive. Can someone verify that?

This is my benchmark, it's running through the entire date range, so maybe a bit overkill...
```cr
require "benchmark"

struct Time
  Benchmark.ips do |bm|
    bm.report "new impl" do
      (1..9999).each do |year|
        ordinal = 0
        (1..12).each do |month|
          (1..days_in_month(year, month)).each do |day|
            ordinal += 1
            Time.absolute_days(year, month, day)
          end
        end
      end
    end
    bm.report "old impl" do
      (1..9999).each do |year|
        ordinal = 0
        (1..12).each do |month|
          (1..days_in_month(year, month)).each do |day|
            ordinal += 1
            Time.absolute_days_old(year, month, day)
          end
        end
      end
    end
  end
end
```

**Comment by RX14** [comment]
@straight-shoota 10% is fine

**Pull request merged**